### PR TITLE
Basic Date Class

### PR DESCRIPTION
**Pull request by Newtsuck**
New Features:
- Added a simple date class with a single function that exposes the
  strftime functon to Loom

**Needs to be tested on the following platforms:**
- [x] Windows
- [x] Android
- [x] OSX
- [x] iOS

### CODE REVIEW

**Comment by bengarney** [comment]
Please ask Miha to test this on device.

File: loom/script/native/core/system/lmDate.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    static const char *formatTime(const char *format)
+    {
+        time_t rawtime;
+        struct tm *timeinfo;
+        char buffer[BUFFER_SIZE];
+
+        time(&rawtime);
+        timeinfo = localtime(&rawtime);
+
+        strftime(buffer, BUFFER_SIZE, format, timeinfo);
```
Comment (bengarney) [inline loom/script/native/core/system/lmDate.cpp]: Should this be BUFFER_SIZE-1? Why or why not?

File: sdk/src/system/Date.ls [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package system {
+    
+/**
+ * Provides native date utility.
+ */
+[Native]
+native class Date {
+    
+    /**
+     * This function wraps the native strftime function from the ctime C++ library.
```
Comment (bengarney) [inline sdk/src/system/Date.ls]: This seems to skip a key detail which is that it seems to only output the current time? Is that the expected behavior?

File: loom/script/native/core/system/lmDate.cpp (hunk 1 above)
Comment (Newtsuck) [inline loom/script/native/core/system/lmDate.cpp]: Good catch

**Comment by SmilyOrg** [comment]
After much turmoil and :rage: I have tested this on OSX and iOS and it appears to work.

**Pull request merged**